source/extensions/clusters/redis/redis_cluster_lb.cc

```diff
@@ -171,9 +171,15 @@ bool isReadRequest(const NetworkFilters::Common::Redis::RespValue& request) {
       first.type() != NetworkFilters::Common::Redis::RespType::BulkString) {
     return false;
   }
-  return NetworkFilters::Common::Redis::SupportedCommands::isReadCommand(first.asString());
+  std::string to_lower_string(first.asString());
+  toLowerTable().toLowerCase(to_lower_string);
+  return NetworkFilters::Common::Redis::SupportedCommands::isReadCommand(to_lower_string);
+}
+
+const ToLowerTable& RedisLoadBalancerContextImpl::toLowerTable() {
+  static auto* table = new ToLowerTable();
```

Comment:
should we make this a static member on the class itself?

Reply:
I did that for the Redis command stats, following the , but the HTTP codec lays forth the pattern Henry is using.

If it's ASCII, which a Redis command should be, you could always use `absl::AsciiStrToLower`.

Reply:
This supports lazy initialization, the static member on the class will be initialized even if there's no calls to the toLowerTable() function.  With the local static variable, it only initialized when it's used.